File: tests/test-sheet.py
import os
import sys
import json
from google.oauth2 import service_account
from googleapiclient.discovery import build

def main():
    # Parse env file manually
    env = {}
    base_dir = os.path.dirname(os.path.abspath(__file__))
    env_path = os.path.join(base_dir, "../.env.local")
    with open(env_path) as f:
        for line in f:
            if "=" in line:
                k, v = line.strip().split("=", 1)
                env[k] = v.strip('"\'')

    creds_info = {
        "type": "service_account",
        "client_email": env["GOOGLE_SERVICE_ACCOUNT_CLIENT_EMAIL"],
        "private_key": env["GOOGLE_SERVICE_ACCOUNT_PRIVATE_KEY"].replace('\\n', '\n'),
        "token_uri": "https://oauth2.googleapis.com/token",
    }
    
    creds = service_account.Credentials.from_service_account_info(
        creds_info, scopes=['https://www.googleapis.com/auth/spreadsheets']
    )
    
    service = build('sheets', 'v4', credentials=creds)
    sheet = service.spreadsheets()
    
    SPREADSHEET_ID = env["GOOGLE_SHEETS_DIRECTORY_ID"]
    
    # Try getting news 
    try:
        result = sheet.values().get(spreadsheetId=SPREADSHEET_ID, range='News').execute()
        print("News Tab contents:")
        print(json.dumps(result.get('values', []), indent=2))
    except Exception as e:
        print(e)
        
if __name__ == '__main__':
    main()

File: tests/test-sheet.mjs
import { google } from 'googleapis';
import fs from 'fs';
import { fileURLToPath } from 'url';
import { dirname, join } from 'path';

const __dirname = dirname(fileURLToPath(import.meta.url));

async function main() {
    const envFile = fs.readFileSync(join(__dirname, '../.env.local'), 'utf8');
    const env = {};
    envFile.split('\n').forEach(line => {
        if (line.includes('=')) {
            const parts = line.split('=');
            env[parts[0].trim()] = parts.slice(1).join('=').trim().replace(/['"]/g, '');
        }
    });

    const auth = new google.auth.GoogleAuth({
        credentials: {
            client_email: env.GOOGLE_SERVICE_ACCOUNT_CLIENT_EMAIL,
            private_key: env.GOOGLE_SERVICE_ACCOUNT_PRIVATE_KEY.replace(/\\n/g, '\n'),
        },
        scopes: ['https://www.googleapis.com/auth/spreadsheets'],
    });

    const sheets = google.sheets({ version: 'v4', auth });
    const response = await sheets.spreadsheets.values.get({
        spreadsheetId: env.GOOGLE_SHEETS_DIRECTORY_ID,
        range: 'News!A2:F',
    });

    fs.writeFileSync(join(__dirname, 'sheet-dump.json'), JSON.stringify(response.data.values, null, 2));
    console.log("Saved to sheet-dump.json!");
}

main().catch(console.error);
